From aa215bbb06ba1d1467fb1d28834608d51a52b293 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Wed, 26 Feb 2020 17:48:16 +0200
Subject: [PATCH 012/230] add Amber accent overlay

	new file:   packages/overlays/AccentColorAmberOverlay/Android.mk
	new file:   packages/overlays/AccentColorAmberOverlay/AndroidManifest.xml
	new file:   packages/overlays/AccentColorAmberOverlay/res/values/colors_device_defaults.xml
	new file:   packages/overlays/AccentColorAmberOverlay/res/values/strings.xml
	modified:   packages/overlays/Android.mk
---
 .../AccentColorAmberOverlay/Android.mk        | 31 +++++++++++++++++++
 .../AndroidManifest.xml                       | 25 +++++++++++++++
 .../res/values/colors_device_defaults.xml     | 22 +++++++++++++
 .../res/values/strings.xml                    | 22 +++++++++++++
 packages/overlays/Android.mk                  |  1 +
 5 files changed, 101 insertions(+)
 create mode 100644 packages/overlays/AccentColorAmberOverlay/Android.mk
 create mode 100644 packages/overlays/AccentColorAmberOverlay/AndroidManifest.xml
 create mode 100644 packages/overlays/AccentColorAmberOverlay/res/values/colors_device_defaults.xml
 create mode 100644 packages/overlays/AccentColorAmberOverlay/res/values/strings.xml

diff --git a/packages/overlays/AccentColorAmberOverlay/Android.mk b/packages/overlays/AccentColorAmberOverlay/Android.mk
new file mode 100644
index 00000000000..98c1f11d61a
--- /dev/null
+++ b/packages/overlays/AccentColorAmberOverlay/Android.mk
@@ -0,0 +1,31 @@
+#
+#  Copyright 2019, The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_RRO_THEME := AccentColorAmber
+
+LOCAL_PRODUCT_MODULE := true
+
+LOCAL_SRC_FILES := $(call all-subdir-java-files)
+
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
+
+LOCAL_PACKAGE_NAME := AccentColorAmberOverlay
+LOCAL_SDK_VERSION := current
+
+include $(BUILD_RRO_PACKAGE)
diff --git a/packages/overlays/AccentColorAmberOverlay/AndroidManifest.xml b/packages/overlays/AccentColorAmberOverlay/AndroidManifest.xml
new file mode 100644
index 00000000000..fd3d7b04410
--- /dev/null
+++ b/packages/overlays/AccentColorAmberOverlay/AndroidManifest.xml
@@ -0,0 +1,25 @@
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.android.theme.color.amber"
+    android:versionCode="1"
+    android:versionName="1.0">
+    <overlay android:targetPackage="android" android:category="android.theme.customization.accent_color" android:priority="1"/>
+
+    <application android:label="@string/accent_color_amber_overlay" android:hasCode="false"/>
+</manifest>
diff --git a/packages/overlays/AccentColorAmberOverlay/res/values/colors_device_defaults.xml b/packages/overlays/AccentColorAmberOverlay/res/values/colors_device_defaults.xml
new file mode 100644
index 00000000000..329e7832e6a
--- /dev/null
+++ b/packages/overlays/AccentColorAmberOverlay/res/values/colors_device_defaults.xml
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <color name="accent_device_default_light">#ff6f00</color>
+    <color name="accent_device_default_dark">#ff6f00</color>
+</resources>
diff --git a/packages/overlays/AccentColorAmberOverlay/res/values/strings.xml b/packages/overlays/AccentColorAmberOverlay/res/values/strings.xml
new file mode 100644
index 00000000000..1e9c788379b
--- /dev/null
+++ b/packages/overlays/AccentColorAmberOverlay/res/values/strings.xml
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+    <!-- Amber accent color name application label. [CHAR LIMIT=50] -->
+    <string name="accent_color_amber_overlay" translatable="false">Amber</string>
+</resources>
diff --git a/packages/overlays/Android.mk b/packages/overlays/Android.mk
index 3fc89eb6b60..efd3fdaaee4 100644
--- a/packages/overlays/Android.mk
+++ b/packages/overlays/Android.mk
@@ -17,6 +17,7 @@ include $(CLEAR_VARS)
 
 LOCAL_MODULE := frameworks-base-overlays
 LOCAL_REQUIRED_MODULES := \
+	AccentColorAmberOverlay \
 	AccentColorBlackOverlay \
 	AccentColorCinnamonOverlay \
 	AccentColorGrayOverlay \
-- 
2.25.1

